### ISSUE TITLE
Binary classifier causes special case bugs due to BinaryLabelizer

### ISSUE THREAD

**Issue by alexjc** [aigamedev/scikit-neuralnetwork#8]
During classification when a Softmax layer is used (recommended), the output probabilities are not calculated correctly due to the BinaryLabelizer in `sklearn` which has special cases for binary classification problems.  In those cases, the output is mapped to 1D rather than 2D for more accurate probability estimation.

There's some code in `predict_proba` to handle this problem, but it needs to be fixed also in `fit()`.

**Comment by ssamot** [comment]
Close by SHA:49c1960699399db7cf91745382a0421292818620
